### ISSUE TITLE
Feature request: Flag to specify output filename

### ISSUE THREAD

**Issue by ad-si** [mixn/carbon-now-cli#30]
```shell
carbon-now --output=./test.png test.js
```

**Comment by ad-si** [comment]
Thanks! Not sure why you choose `--target` over `--output`, as all CLI tools I can currently think of use '--output', but nevermind 😅

**Comment by ad-si** [comment]
Also I would have used `-t test.png` instead of `-t test`. Now if someone reads that piece of code in a script they have no idea what it means.
`-t test` could mean anything: Run tests before generating the image, execute it in a special test mode.
Also you often safe file names in a variable like `$OUTPUT_PATH`. With your inteface one has now to laboriously split up the variable in its parts. If you have the parts already and want to join them, however, it's much easier: `--output $directory/$filename`

Long story short: You should support `--output ~/Desktop/test.png` for absolute paths and `--output test.png` for relative paths. 😉

**Comment by ad-si** [comment]
And if someone tries to `--output ~/Desktop/test.jpeg` simply throw an Error `The jpeg format is currently not supported`

**Comment by mixn** [comment]
> Thanks! Not sure why you choose `--target` over `--output`, as all CLI tools I can currently think of use '--output', but nevermind 😅

Because `-o` is already taken by `--open` and the shorthand should match the beginning letter imo.

> You should support `--output ~/Desktop/test.png` for absolute paths and `--output test.png` for relative paths. 😉

Note taken. 👍😊 For now `-l` + `-t` will suffice.

**Comment by ad-si** [comment]
> Because -o is already taken by --open

Open should be a subcommand and not a flag, but that's another topic 😅

> the shorthand should match the beginning letter imo

This tool shouldn't have any shortcuts. It's not that anyone will use it so often (unlinke e.g. `ls`) that typing a few characters more would be a problem. However, it will be used in scripts and then people with stumble upon cryptic `-o,-s,-e,-i,-l,-t, -o, -p,-h` flags in those scripts and have no idea what they mean.

> For now -l + -t will suffice.
If you don't change it now, you won't be able to change it in the future due to backwards incompatibility problems 😞

**Comment by mixn** [comment]
> Open should be a subcommand and not a flag, but that's another topic 😅

This is my first OS CLI and your passive-aggressiveness is not helping. So my thoughts on this are: ¯\_(ツ)_/¯ 😄 

> This tool shouldn't have any shortcuts. It's not that anyone will use it so often (unlinke e.g. `ls`) that typing a few characters more would be a problem. However, it will be used in scripts and then people with stumble upon cryptic `-o,-s,-e,-i,-l,-t, -o, -p,-h` flags in those scripts and have no idea what they mean.

A CLI will always be used more frequently in the CL than in scripts, hence the shortcuts. Also, feel free to use the `--long-version` in scripts, nothing wrong with that. 👍

**Comment by ad-si** [comment]
> This is my first OS CLI

Oops, sorry. You're profile sounded like you'd be an experienced developer ...

>  feel free to use the --long-version in scripts, nothing wrong with that.

Yeah, I will. But others won't. They will simply copy what they typed and not bother about making it readable and others have to deal with it 😭

**Comment by mixn** [comment]
> Oops, sorry. You're profile sounded like you'd be an experienced developer ...

We’re all just beginners, really. We never stop learning. And just cause I’m a JS consultant and developer, doesn’t mean I know all of UNIX by heart since I am 5 y/o. And fairly, that is **fine**. :) 

Maybe you can teach us all, Adrian? 😄 Feel free to, we’d all appreciate it. Contribute your learnings and make tutorials on it, I’m glad to dive in and share it.

I’m done with this thread and your ambiguous writing. :)

**Comment by ad-si** [comment]
That's what I'm trying. Feel free to check it out at: adriansieber.com

I actually just created a TODO: "Write blog post about best practices for command line interfaces" :grin:

**Comment by mixn** [comment]
> That's what I'm trying. Feel free to check it out at: adriansieber.com

I’ve seen your website, it’s cool that you have one! :) Looking forward to that post!
